lib/src/Pos/Lrc/DB/Leaders.hs

```diff
@@ -17,23 +19,39 @@ import           Universum
 import           Pos.Binary.Class      (serialize')
 import           Pos.Binary.Core       ()
 import           Pos.Context.Functions (genesisLeaders)
-import           Pos.Core              (EpochIndex, HasConfiguration, SlotLeaders)
+import           Pos.Core              (EpochIndex, HasConfiguration,
+                                        HasProtocolConstants,
+                                        LocalSlotIndex (UnsafeLocalSlotIndex),
+                                        SlotId (SlotId), SlotLeaders, StakeholderId,
+                                        flattenSlotId)
 import           Pos.DB.Class          (MonadDB, MonadDBRead)
-import           Pos.Lrc.DB.Common     (getBi, putBi)
+import           Pos.Lrc.DB.Common     (getBi, putBatchBi, putBi)
 
 ----------------------------------------------------------------------------
 -- Getters
 ----------------------------------------------------------------------------
 
-getLeaders :: MonadDBRead m => EpochIndex -> m (Maybe SlotLeaders)
-getLeaders = getBi . leadersKey
+getLeadersForEpoch :: MonadDBRead m => EpochIndex -> m (Maybe SlotLeaders)
+getLeadersForEpoch = getBi . leadersForEpochKey
+
+getLeader :: MonadDBRead m => SlotId -> m (Maybe StakeholderId)
+getLeader = getBi . leaderKey
 
 ----------------------------------------------------------------------------
 -- Operations
 ----------------------------------------------------------------------------
 
-putLeaders :: MonadDB m => EpochIndex -> SlotLeaders -> m ()
-putLeaders epoch = putBi (leadersKey epoch)
+-- | Put leaders for all slots in the specified epoch in the DB.
+-- The DB contains two mappings:
+-- * EpochIndex -> SlotLeaders
+-- * SlotId -> StakeholderId (added in CSE-240)
+putLeadersForEpoch :: MonadDB m => EpochIndex -> SlotLeaders -> m ()
+putLeadersForEpoch epoch leaders = do
+    putLeadersForEpochAllAtOnce epoch leaders
```

Comment:
What happens if I power off pc between these two lines?

Reply:
Good catch @volhovM. 
You can pull up those helpers below into the local function scope using _where_, change them to return the DB operation and perform `BatchOp` on the list containing both of their operations. Or `putBatchBi`.

Reply:
Actually nothing bad will happen, but I would prefer doing both lines atomically.